### PR TITLE
Green update from 0.9.2

### PR DESCRIPTION
**Pull request by simonschaufi**
make smooth upgrade from 0.9.3 (fresh install)

FIX:
- path to installPermissions
- all issues with keys

Related: #517 but does not solve the issues mentioned in this ticket!

### CODE REVIEW

File: updater/updater.php [hunk 1]
```diff
@@ -925,24 +924,21 @@
 DROP `effort`,
 DROP `approved`
 ;", false);
-
-    exec_query("ALTER TABLE `${p}projects_activities`
-ADD `budget`     decimal(10,2) NOT NULL DEFAULT '0.00',
-ADD `effort`     decimal(10,2) DEFAULT NULL,
-ADD `approved`   decimal(10,2) DEFAULT NULL
-;", false);
```
Comment (simonschaufi) [inline updater/updater.php:933]: this was removed on purpose!

File: updater/updater.php [hunk 2]
```diff
@@ -644,7 +640,7 @@
 
     $result = $database->queryAll("SELECT usr_ID,usr_grp FROM ${p}usr");
     foreach ($result as $row) {
-        exec_query("INSERT INTO ${p}grp_usr (`grp_ID`,`usr_ID`) VALUES($row[usr_grp],$row[usr_ID]);");
+        exec_query('INSERT INTO ' . $p . 'grp_usr (`grp_ID`, `usr_ID`) VALUES(' . $row['usr_grp'] . ',' . $row['usr_ID'] . ');');
```
Comment (kevinpapst) [inline updater/updater.php:643]: please follow the same pattern as all other statements

File: updater/updater.php (hunk 1 above)
Comment (kevinpapst) [inline updater/updater.php:933]: Why? From the comment I read that the columsn will purposely be removed and then re-added.
Now they are only removed... please explain :)

File: updater/updater.php [hunk 3]
```diff
@@ -925,24 +924,21 @@
...
 
 if ((int)$revisionDB < 1373) {
+    Kimai_Logger::logfile("-- update to r1373");
     exec_query("ALTER TABLE `${p}activities` DROP `assignable`;");
 }
 
 if ((int)$revisionDB < 1374) {
-    require("installer/installPermissions.php");
+    Kimai_Logger::logfile("-- update to r1374");
+    require("../installer/installPermissions.php");
```
Comment (kevinpapst) [inline updater/updater.php:941]: can we use a **DIR** constant here, to make 100% sure it will be loaded?

File: updater/updater.php [hunk 4]
```diff
@@ -1113,6 +1131,4 @@
 
 Kimai_Logger::logfile("-- update finished --------------------------------");
 
-// ================================================================================
```
Comment (kevinpapst) [inline updater/updater.php:1116]: Where is all the ASCII art gone to? ;-)

File: updater/updater.php [hunk 5]
```diff
@@ -1017,40 +1019,53 @@
...
+    exec_query("ALTER TABLE `${p}users` ADD UNIQUE  KEY(`apiKey`);", false);
 
     exec_query("UPDATE `${p}preferences` SET `option` = 'ui.project_comment_flag' WHERE `option` = 'ui.pct_comment_flag';");
 }
 
 if ((int)$revisionDB < 1382) {
     Kimai_Logger::logfile("-- update to r1382");
-    exec_query("ALTER TABLE `${p}membershipRoles` ADD `core-user-view` tinyint DEFAULT 0 AFTER `core-user-unassign`;", false);
+    #column already added in installer/installPermissions.php in r1374
+    #exec_query("ALTER TABLE `${p}membershipRoles` ADD `core-user-view` tinyint DEFAULT 0 AFTER `core-user-unassign`;", false);
```
Comment (kevinpapst) [inline updater/updater.php:1068]: Yes, but installPermissions is not executed in this step! And if it was added later on, people will miss this column.

File: updater/updater.php [hunk 6]
```diff
@@ -1017,40 +1019,53 @@
 if ((int)$revisionDB < 1381) {
     Kimai_Logger::logfile("-- update to r1381");
     // make sure all keys are defined correctly
-    exec_query("ALTER TABLE `${p}activities`          ADD PRIMARY KEY(`activityID`);", false);
-    exec_query("ALTER TABLE `${p}configuration`       ADD PRIMARY KEY(`option`);", false);
-    exec_query("ALTER TABLE `${p}customers`           ADD PRIMARY KEY(`customerID`);", false);
-    exec_query("ALTER TABLE `${p}expenses`            ADD PRIMARY KEY(`expenseID`);", false);
+    # primary key since r733
```
Comment (kevinpapst) [inline updater/updater.php:1022]: This line and following: why did you remove them?

File: updater/updater.php (hunk 1 above)
Comment (simonschaufi) [inline updater/updater.php:933]: They are removed from the **activities** table and added on the **projects_activities** table. Thats why this migration will throw an error on duplicate columns since they exist since [r1368](https://github.com/kimai/kimai/blob/master/updater/updater.php#L813-L821) already. I didn't find the revision where these columns should have been removed.

File: updater/updater.php (hunk 6 above)
Comment (simonschaufi) [inline updater/updater.php:1022]: made a comment above: primary key since r733. the columns just get renamed and therefor still have the primary keys.

File: updater/updater.php (hunk 4 above)
Comment (simonschaufi) [inline updater/updater.php:1116]: too much of it makes the code ugly :) For the footer it is really not needed.

File: updater/updater.php (hunk 5 above)
Comment (simonschaufi) [inline updater/updater.php:1068]: installPermissions is run within one release (0.9.3). I made sure it is not run in 0.9.2. So this column gets added twice in 0.9.3 and if people are always on master, they already have it ;) So this can be safely removed.

File: updater/updater.php (hunk 3 above)
Comment (simonschaufi) [inline updater/updater.php:941]: can you modify my pull request and add it? you now have the permission to do so ;)

File: updater/updater.php (hunk 2 above)
Comment (simonschaufi) [inline updater/updater.php:643]: I actually want to get rid of variables in strings because of [this](https://github.com/php-fig-rectified/fig-rectified-standards/blob/master/PSR-2-R-coding-style-guide-additions.md#strings-and-concatination). I know this is not PSR-2 but it just makes the code more readable imho.
